gdnative-core/src/core_types/variant.rs

```diff
@@ -109,6 +112,13 @@ macro_rules! decl_variant_type {
             )*
         }
 
+        impl VariantType {
+            /// The potential names of VariantTypes. Mostly used for serialization.
+            pub const NAMES: &'static [&'static str] = &[
```

Comment:
Should `NAMES` be available even when the `serde` feature is disabled? (not saying it shouldn't, but to make sure this is a conscious decision). Same for items below.

Reply:
IMO, there's no reason disable it if `serde` is disabled. Who knows if someone might end up using it for something else, and IMO the only reason to even make `serde` a feature in the first place is to avoid compiling `serde` itself.